core/src/main/java/hudson/FilePath.java

```diff
@@ -1962,7 +1963,7 @@ public void renameTo(final FilePath target) throws IOException, InterruptedExcep
         act(new SecureFileCallable<Void>() {
             private static final long serialVersionUID = 1L;
             public Void invoke(File f, VirtualChannel channel) throws IOException {
-            	reading(f).renameTo(creating(new File(target.remote)));
+                Files.move(reading(f).toPath(), creating(new File(target.remote)).toPath(), StandardCopyOption.COPY_ATTRIBUTES, LinkOption.NOFOLLOW_LINKS);
```

Comment:
I don't try/catch here as `Files.move()` throws an `IOException`, which is what we would like to propagate out from this class.

Reply:
That's fine, but you do want to merge with master and use `Util#fileToPath` anywhere you use `File#toPath` so that the runtime exception `InvalidPathException` is wrapped as an `IOException`.

Reply:
Also, the [documentation for Files#move](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#move-java.nio.file.Path-java.nio.file.Path-java.nio.file.CopyOption...-) doesn't include `StandardCopyOption.COPY_ATTRIBUTES`, so I'd remove it here. (I think most implementations of `Files#move` attempt to copy attributes by default.)